src/functions/assertions/HaveParameter.ps1

```diff
@@ -71,20 +73,30 @@
         }
 
         foreach ($parameter in $parameters) {
+            if ($Name -ne $parameter.Name.VariablePath.UserPath) {
+                continue
+            }
+
             $paramInfo = [PSCustomObject] @{
                 Name             = $parameter.Name.VariablePath.UserPath
                 Type             = "[$($parameter.StaticType.Name.ToLower())]"
+                HasDefaultValue  = $false
                 DefaultValue     = $null
                 DefaultValueType = $parameter.StaticType.Name
             }
 
+            # Default value here contains a descriptor object of the default value,
+            # so this is null only when default value is not present at all, if default value
+            # is actually $null, this will have an object describing the type and the $null value.
             if ($null -ne $parameter.DefaultValue) {
-                if ($parameter.DefaultValue.PSObject.Properties['Value']) {
-                    $paramInfo.DefaultValue = $parameter.DefaultValue.Value
-                }
-                else {
-                    $paramInfo.DefaultValue = $parameter.DefaultValue.Extent.Text
-                }
+                # The actual value of the default value can be falsy (e.g. $null, $false or 0)
+                # use this flag to communicate if default value was found in the AST or not,
+                # no matter if the actual default value is falsy.
+                # That is: param($param1 = $false) will set this to true for $param1
+                # but param($param1) will have this set to false, because there was no default value.
+                $paramInfo.HasDefaultValue = $true
+                # $null does not have a Value
+                $paramInfo.DefaultValue = if ($null -ne $parameter.DefaultValue.Value) { $parameter.DefaultValue.Value } else { $parameter.DefaulValue.Extent }
```

Comment:
This is problematic, because we are lifting the value from AST it comes either as a literal when it is a constant (e.g. 0), but as semantic tree when it is not a literal, e.g. $true in the simplest case. Which is a variable descriptor. 

I think this will be problematic for variables, and also for other default values that need evaluation, we probably don't want to evaluate them ourselves.